### PR TITLE
suit: use SUIT boot by default for nrf54h20

### PR DESCRIPTION
**Pull request by robertstypa**
Use SUIT to boot Lilium samples

Ref: NCSDK-27372

Signed-off-by: Robert Stypa <robert.stypa@nordicsemi.no>

### CODE REVIEW

**Comment by lemrey** [comment]
Please remember to update the commit title to something like `suit: ` before merging :)

**Review by tomchy — APPROVED** [review]

File: cmake/sysbuild/suit.cmake [hunk 1]
```diff
@@ -165,6 +165,11 @@ function(suit_create_package)
 
   suit_set_absolute_or_relative_path(${SB_CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION} ${PROJECT_BINARY_DIR} INPUT_TEMPLATES_DIRECTORY)
   set(ENVELOPE_SHALL_BE_SIGNED ${SB_CONFIG_SUIT_ENVELOPE_SIGN})
+  if(NOT DEFINED west_realpath)
+    # Twister build - ignore and store in the build folder!
+    set(SB_CONFIG_SUIT_ENVELOPE_EDITABLE_TEMPLATES_LOCATION "./")
+  endif()
+  sysbuild_get(ENVELOPE_SHALL_BE_SIGNED IMAGE ${DEFAULT_IMAGE} VAR CONFIG_SUIT_ENVELOPE_SIGN KCONFIG)
```
Comment (e-rk) [inline cmake/sysbuild/suit.cmake:172]: Not for this PR, but how about copying the templates into `build/<sample-name>/...` directory? The flow would be as follows:
1. If template exists at `<sample-dir>`, then copy it to `build/<sample-name>/...` and use it to generate the envelope
2. If template does not exist at `<sample-dir>`, then copy it from default location into `<sample-dir>` and then do action from 1.

**Review by e-rk — APPROVED** [review]

File: cmake/sysbuild/suit.cmake (hunk 1 above)
Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:172]: I'll create a JIRA ticket to discuss that feature request

Comment (robertstypa) [reply cmake/sysbuild/suit.cmake:172]: https://nordicsemi.atlassian.net/browse/NCSDK-27662

File: sysbuild/Kconfig.suit [hunk 2]
```diff
@@ -6,9 +6,9 @@
 
 menu "SUIT"
 
-# 'default y if SOC_SERIES_NRF54HX' to be added when SUIT is to be enabled globally.
 menuconfig SUIT_ENVELOPE
 	bool "Create SUIT envelope"
+	default y if SOC_SERIES_NRF54HX
```
Comment (ahasztag) [inline sysbuild/Kconfig.suit:11]: @e-rk will this influence flash_companion? Shouldn't the flash_companion app set `SUIT_ENVELOPE` to `n`?

Comment (e-rk) [reply sysbuild/Kconfig.suit:11]: No, this is the sysbuild Kconfig that must be enabled always with or without the flash companion.

**Review by ahasztag — APPROVED** [review]

**Pull request merged**